### ISSUE TITLE
Fatal IllegalStateException on launch from reading item data

### ISSUE THREAD

**Issue by TsukiTheDwagon** [LifeIsAParadox/Skyblocker#38]
Crash log for reference: [Pastebin](https://pastebin.com/BEfFJPbp)

From what it looks like, Skyblocker instantly crashes on load with an IllegalStateException if some sort of null value is assigned to any key in an item's JSON data in the NEU repo.

From reading the crash log, it seems that the function importItemFiles() in ItemRegistry.java is at fault, and I've somewhat isolated the issue to these lines of code here:
```Java
try {
  String fileContent = Files.readString(Paths.get(path));
  jsonObjs.add(JsonParser.parseString(fileContent).getAsJsonObject()); // line that causes the crash
} catch (IOException e) { // line that could be changed for a possible fix
  System.err.println("Couldn't import " + path);
  e.printStackTrace();
}
```
Probably the best solution would be either to add this specific exception to the list of exceptions to catch, or widen the scope of the exceptions altogether (e.g. to just a generic Exception), that way it simply returns an error for the specific item that had a null tag instead of fatally crashing Minecraft altogether.

I'm not 100% sure which option would fit the codebase better, so I've just left it open as an issue instead of pull requesting it.

**Comment by KonaeAkira** [comment]
Sorry, I couldn't replicate this. Did you modify the repo uri or the local repo itself?

**Comment by TsukiTheDwagon** [comment]
I just installed the mod, nothing else. I can try loading it on a fresh MultiMC instance in a bit when I get to my PC.

**Comment by TsukiTheDwagon** [comment]
Ok, after doing a bit of debugging on my main instance folder after seeing a fresh insteance open just fine, it seems the mod created a corrupted item repo folder sometime in the past when I installed it and it crashed on me before. Simply deleting .minecraft/config/skyblocker/item-repo seems to have fixed the issue.

I don't think I modified it myself or anything, it seems it was just a failed previous install's leftover remnants that caused it to break. Probably some sort of merge conflict with git checkout looking at the logs.

Not sure why the crash log returned an exception that wasn't neccesarily related to the root of the problem like that, but I guess it's fixed now!
